### PR TITLE
Azure attachment plugin

### PR DESCRIPTION
**Pull request by wassoares**
This pull request implements the attachment plugin for the Azure cloud.

### Dependencies
Requires that the following branches have already merged with develop branch:
- azure-volume-plugin - PR #417;

### Additional information
The unit tests were implemented in another branch.

### CODE REVIEW

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    AttachmentInstance buildAttachmentInstance(Disk disk) {
+        DiskInner diskInner = disk.inner();
+        String id = diskInner.id();
+        String cloudState = disk.isAttachedToVirtualMachine() 
+                ? AzureStateMapper.ATTACHED_STATE
+                        : AzureStateMapper.UNATTACHED_STATE;
+        
+        String computeId = disk.virtualMachineId();
+        String volumeId = disk.id();
+        String device = null;
```
Comment (fgan1) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java:127]: Use the constant dedicated to this kind of value(without information). 
Look at this link: https://github.com/fogbow/resource-allocation-service/blob/azure-attachment-plugin/src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureGeneralUtil.java#L10

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    @VisibleForTesting
+    VirtualMachine doGetVirtualMachineSDK(Azure azure, String virtualMachineId) throws FogbowException {
+        return AzureVirtualMachineSDK
+                .getVirtualMachine(azure, virtualMachineId)
+                .orElseThrow(() -> new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND));
+    }
+
+    @VisibleForTesting
+    String buildResourceId(String subscriptionId, String resourceName) {
```
Comment (fgan1) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java:156]: This method seems strange to me. You are using this same method to create VirtualMachineId and DiskId? Is it right? I think that It just works for the DiskId case.

**Comment by fgan1** [comment]
Well done!

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java (hunk 1 above)
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java:127]: Ok, I'll fix it.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java (hunk 2 above)
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java:156]: In fact, I don't understand how the system test worked, but I will fix it right now.

**Pull request merged**